### PR TITLE
Various small cleanups and bugfixes

### PR DESCRIPTION
**Pull request by jcupitt**
A few smallish things I found while getting the openslide connnector working.

### CODE REVIEW

**Comment by jcupitt** [comment]
Sorry, the history on this is a bit of a mess.

Summary:

1. I needed to be able to get tags from keywords, so I added another hash. I named the keyword <-> tag functions as :c:func:`dcm_dict_tag_from_keyword()` and :c:func:`dcm_dict_keyword_from_tag()`.
2. I thought the `_lookup` names were a little unclear (it's not obvious which direction they are going) so I renamed the other similar functions for consistency.
3. We had the function `dcm_filehandle_read_filehandle_meta()`, left over from when we renamed `file` as `filehandle`. I thought this was probably an error and it should be `dcm_filehandle_read_file_meta()`.
4. We used `_meta()` in some places and `_metadata()` in others, which seemed inconsistent, so I switched to `_metadata()` everywhere.
5. I found an annoying bug in dataset_insert.

File: include/dicom/dicom.h [hunk 1]
```diff
@@ -1443,11 +1457,11 @@ DcmFilehandle *dcm_filehandle_create_from_memory(DcmError **error,
  * :param error: Error structure pointer
  * :param filehandle: File
  *
- * :return: File Metainformation
+ * :return: File Metadata
  */
 DCM_EXTERN
-DcmDataSet *dcm_filehandle_read_filehandle_meta(DcmError **error, 
-                                                DcmFilehandle *filehandle);
+DcmDataSet *dcm_filehandle_read_file_metadata(DcmError **error,
```
Comment (hackermd) [inline include/dicom/dicom.h:1463]: This is not the metadata of the data set (i.e., the collection of non-bulkdata elements), but the [File Meta Information](https://dicom.nema.org/medical/dicom/current/output/chtml/part10/chapter_7.html#sect_7.1). Please change back to "meta". That name was chosen on purpose.

**Review by hackermd — CHANGES_REQUESTED** [review]

**Comment by hackermd** [comment]
@jcupitt One other thing that I just noted, which is not directly related to this PR: The `_DcmVR` enum not only contains individual VRs, but also options such as "OB or OW" (see [here](https://github.com/ImagingDataCommons/libdicom/blob/39815afef378118c15084edabc72b70bbfd6d0c7/include/dicom/dicom.h#L226-L229)). Strictly speaking, those are not VRs and, in my opinion, therefore don't belong into the enum for the type definition. Can we handle those options differently?

File: include/dicom/dicom.h (hunk 1 above)
Comment (jcupitt) [reply include/dicom/dicom.h:1463]: OK, I'll change it to `dcm_filehandle_read_file_meta()`

**Comment by jcupitt** [comment]
I split the `DcmVR` enum into two, a public and a private one, and put the combination VRs into the private version.

There are two new things:

```
bool dcm_is_valid_vr_for_tag(DcmVR vr, uint32_t tag);
DcmVR dcm_vr_from_tag(uint32_t tag);
```

That can quickly check if a VR can be used for a known tag, and for implicit mode, a thing that can check that a tag has a unique VR.

**Review by hackermd — APPROVED** [review]

File: include/dicom/dicom.h [hunk 2]
```diff
@@ -858,7 +858,7 @@ bool dcm_element_set_value_sequence(DcmError **error,
  * :param indentation: Number of white spaces before text
  */
 DCM_EXTERN
-void dcm_element_print(const DcmElement *element, uint8_t indentation);
+void dcm_element_print(const DcmElement *element, int indentation);
```
Comment (hackermd) [inline include/dicom/dicom.h:861]: We wanted to use exact-width integer types in the public API. What is the motivation for switching to `int`?

Comment (jcupitt) [reply include/dicom/dicom.h:861]: Sized int types are useful when variables are derived from a file format with specific numeric size. For example, DICOM elements have a length in a four byte unsigned value, so `uint32_t` is meaningful as the return value for  `dcm_element_get_length()`.

`indentation` however is an abstract value and does not relate to any concrete representation. Giving it specific format adds complexity for no benefit, and I think the extra complexity actually makes bugs more likely.

**Comment by bgilbert** [comment]
Looks like tests haven't run.  A re-push might fix it, but I think it can also happen if GitHub fails to create a merge commit to run tests against.

**Comment by jcupitt** [comment]
Oh, that's odd, it was showing all all complete for me yesterday. I'll push again.

**Pull request closed without merging**